app/components/insights-date-display.js

```diff
@@ -0,0 +1,32 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import moment from 'moment';
+import { DEFAULT_INSIGHTS_INTERVAL } from 'travis/services/insights';
+
+export default Component.extend({
+  tagName: 'span',
+  classNames: ['insights-dates'],
+
+  insights: service(),
+
+  interval: DEFAULT_INSIGHTS_INTERVAL,
+
+  dates: computed('interval', function () {
+    return this.insights.getDatesFromInterval(this.interval);
+  }),
+
+  startDate: computed('dates', function () {
+    if (this.dates.length === 2) {
+      return moment(this.dates[0]).format('MMMM DD, YYYY');
+    }
+    return '...';
```

Comment:
You could use `dates.firstObject` and `date.lastObject` here:

```javascript
startDate: computed('dates.firstObject', function () {
  const { dates = [] } = this;
  return dates.firstObject ? moment(dates.firstObject).format('MMMM DD, YYYY') : '...';
})
```

Reply:
Otherwise, the property should depend on `dates.[]` - because it depends not only on `dates` property itself, but also on its content.

Reply:
Ahh cool, that makes sense! Updated to use `dates.firstObject`